### PR TITLE
Batchnorm inference instances, external API, client examples and gtests

### PR DESCRIPTION
**Pull request by qianfengz**
This P.R provides:
1. Instances of `DeviceElementwiseImpl` for `batchnorm-infer`
2. External API to collect all instances for batchnorm-Infer
3. Client example to use the example API and Device Op API to do `batchnorm-infer`
4. Gtests to cover a wile range of `batchnorm-infer` instances

- This P.R set an example for how to use generic C.K device op (`element-wise` op on 1d tensors) to implement specific functions (eg. `batchnorm-infer`)
- This P.R also provides example on using  dimension expanding on tensor by using 0-stride value on the added dimensions. The applications need to expand the dimensions of `scale, bias, mean, variance` from 1 to 4, enable their lengths the same as `xyLengths`, since the `DeviceElementwise` API assumes all the inputs have the same dimensions

### CODE REVIEW

File: include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp
Comment (asroy) [inline include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp:20]: I think this should be implemented as elementwise operation

File: include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp
Comment (asroy) [inline include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp:57]: Is this class really necessary?

File: include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp
Comment (qianfengz) [reply include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp:20]: Do you mean move this to the file `include/ck/tensor_operation/gpu/element/element_wise_operation.hpp` ?

File: include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp
Comment (qianfengz) [reply include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp:57]: I think this definition make the `GetInstances()` API easier to understand,  since the user does not have to use `Tuple` to construct the instance of  a template class of `DeviceElementwise`

File: include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp
Comment (asroy) [reply include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp:20]: Yes, I mean moving it into ```element_wise_operation.hpp```

File: include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp
Comment (asroy) [reply include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp:57]: I think this is not necessary. If you can use the base class for doing the same thing, then the derived class is not needed.

If you think the user may not know how to use the baes class for BN-infer, you can write example for that.

**Review by asroy — CHANGES_REQUESTED** [review]

File: include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp
Comment (qianfengz) [reply include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp]: Solved by commit [67203](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/531/commits/9b04b3b920de0195a522a5cef735287283a67203)

File: include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp
Comment (qianfengz) [reply include/ck/tensor_operation/gpu/device/device_elementwise_extension.hpp]: Solved by commit [dacd8](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/531/commits/31da74c1ac36558339455d578635a24b881dacd8)

**Review by asroy — APPROVED** [review]
@qianfengz this PR can be merged once the merge conflict is resolved

**Comment by qianfengz** [comment]
> @qianfengz this PR can be merged once the merge conflict is resolved

Done

**Pull request merged**
